### ISSUE TITLE
Staging Review finding: Typeahead suggested results not announced to screen reader users

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#84584]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Contention Classification
**Product:** Disability Comp Application (21-526 EZ)
**Feature:** Improve conditions submission page

## Findings details
**VA.gov Experience Standard - issue:** [User can't perceive an element.](https://design.va.gov/about/experience-standards/#user-can-perceive-all-meaningful-elements)
**VA.gov Experience Standard - category:** [Comprehension](https://design.va.gov/about/experience-standards/#category-comprehension)
**Launch-blocking:** Yes
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

Screen readers are not announcing the typeahead search suggestions when adding or editing a condition. As you arrow down through the suggested results, the text that was entered into the input is announced for each item.

See the video in the collapsed section below for a demo.

<details><summary>Link, screenshot or steps to recreate</summary>

https://github.com/department-of-veterans-affairs/va.gov-team/assets/6844775/88ea8c1a-d739-41cf-a24f-fb9f6ae1f8b2

For comparison, here's the current typeahead behavior of the search component in the VA design system:

https://github.com/department-of-veterans-affairs/va.gov-team/assets/6844775/b455b6fb-f337-48c0-9d45-fb40b09ff7f5

</details>

## Recommended action

Ideally, I'd like the conditions search component to behave as close to that as possible. Note that the design system component includes several ARIA attributes that aren't present in the 526ez conditions input: [aria-autocomplete="list"](https://developer.mozilla.org/en-US/docs/web/Accessibility/ARIA/Attributes/aria-autocomplete), [aria-controls="va-search-listbox"](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls), [aria-expanded=[true/false]](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded), [aria-haspopup="listbox"](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup), [role="combobox"](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Combobox_Role), and [aria-activedescendant=[currently highlighted descendent]](https://developer.mozilla.org/en-US/docs/web/Accessibility/ARIA/Attributes/aria-activedescendant).

```
<input class="usa-input" 
    id="search-field" 
    name="search" 
    type="search" 
    aria-autocomplete="list" 
    aria-label="Search" 
    autocomplete="off" 
    aria-controls="va-search-listbox" 
    aria-expanded=[true/false] 
    aria-haspopup="listbox" 
    role="combobox" 
    aria-activedescendant="listbox-option-[index]">
```

Typeahead search behavior is incredibly tricky to communicate in a useful way to screen reader users, and all of these ARIA attributes work together to ensure that the functionality and state of the component all gets communicated clearly. Strongly recommend you follow the design system component as closely as possible.

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** N/A
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by alatempa24** [comment]
Closing as complete per merge 30241
